### PR TITLE
Added minimum memory limit in restricted process executor and submission executor interface

### PR DESCRIPTION
**Pull request by gogo4ds**
Add tests for Judge Server
https://github.com/SoftUni-Internal/interactive-issues/issues/90

### CODE REVIEW

File: OJS.Workers.SubmissionProcessors/SubmissionExecutor.cs [hunk 1]
```diff
@@ -6,19 +6,19 @@
...
-            OjsSubmission<TInput> submission)
+            IOjsSubmission submission)
             where TResult : ISingleCodeRunResult, new()
         {
             var executionStrategy = this.CreateExecutionStrategy(submission);
 
-            var executionContext = this.CreateExecutionContext(submission);
+            var executionContext = this.CreateExecutionContext<TInput>(submission);
 
             return this.ExecuteSubmission<TInput, TResult>(executionStrategy, executionContext, submission);
```
Comment (achobanov) [inline OJS.Workers.SubmissionProcessors/SubmissionExecutor.cs]: Long line.

File: OJS.Workers.SubmissionProcessors/SubmissionExecutor.cs [hunk 2]
```diff
@@ -39,11 +39,12 @@ private IExecutionStrategy CreateExecutionStrategy(IOjsSubmission submission)
             }
         }
 
-        private IExecutionContext<TInput> CreateExecutionContext<TInput>(OjsSubmission<TInput> submission)
+        private IExecutionContext<TInput> CreateExecutionContext<TInput>(IOjsSubmission submission)
```
Comment (achobanov) [inline OJS.Workers.SubmissionProcessors/SubmissionExecutor.cs]: Long line.

File: OJS.Workers.SubmissionProcessors/SubmissionExecutor.cs (hunk 1 above)
Comment (gogo4ds) [reply OJS.Workers.SubmissionProcessors/SubmissionExecutor.cs]: not more than 120 characters

File: OJS.Workers.SubmissionProcessors/SubmissionExecutor.cs (hunk 2 above)
Comment (gogo4ds) [reply OJS.Workers.SubmissionProcessors/SubmissionExecutor.cs]: not more than 120 characters

**Pull request merged**